balancer.go

```diff
@@ -89,6 +86,18 @@ func (lb *LeastBytes) Balance(msg Message, partitions ...int) int {
 	return c.partition
 }
 
+func (lb *LeastBytes) partitionChanged(partitions ...int) bool {
+	if len(partitions) != len(lb.counters) {
+		return true
+	}
+	for _, p := range partitions {
```

Comment:
if it's true that partitions are always passed in 0-indexed and monotonically increasing with no gaps, i think it's possible we only need the length check that you added (and don't need to iterate `partitions` or the `counterOf` function anymore), thoughts?

Reply:
Yes, you 're right.  Length checking is enough.  
May I remove the counterOf function in this pr, because it 's not used?